core/src/main/java/nl/nn/adapterframework/pipes/TimeoutGuardPipe.java

```diff
@@ -97,21 +60,20 @@ public PipeRunResult doPipe(Message message, IPipeLineSession session) throws Pi
 			timeout_work = Integer.valueOf(timeout_work_str);
 		}
 
-		DoPipe doPipe = new DoPipe(message, session, Thread.currentThread().getName(), ThreadContext.peek());
-		ExecutorService service = Executors.newSingleThreadExecutor();
-		Future<Message> future = service.submit(doPipe);
-		Object result = null;
+		log.debug(getLogPrefix(session) + "setting timeout of [" + timeout_work + "] s");
+		TimeoutGuard tg = new TimeoutGuard(timeout_work, getName()) {
+			@Override
+			protected void kill() {
+				throw new IllegalStateException("exceeds timeout of [" + timeout_work + "] s, interupting");
+			}
```

Comment:
In deze kill() moet ook het eigenlijke werk afgebroken worden. Je moet  op een of andere manier hebben om in descender classes dat te kunnen implementeren. Anders is deze aanpassing van de regen in de drup